packages/api/src/promise/Api.ts

```diff
@@ -249,4 +249,73 @@ export default class ApiPromise extends ApiBase<'promise'> {
       }) as UnsubscribePromise;
     } as StorageEntryPromiseOverloads;
   }
+
+  protected exposeRpc (rpc: DecoratedRpc<'rxjs'>): DecoratedRpc<'promise'> {
+    return Object.keys(rpc).reduce((result, _sectionName) => {
+      const sectionName = _sectionName as keyof DecoratedRpc<'rxjs'>;
+
+      result[sectionName] = Object.keys(rpc[sectionName]).reduce((section, methodName) => {
+        const method = rpc[sectionName][methodName];
+        section[methodName] = this.exposeMethod(method);
+
+        return section;
+      }, {} as DecoratedRpc$Section<'promise'>);
+
+      return result;
+    }, {} as DecoratedRpc<'promise'>);
+  }
+
+  protected exposeMulti (multi: QueryableStorageMulti<'rxjs'>): QueryableStorageMulti<'promise'> {
+    return this.exposeMethod(multi) as QueryableStorageMulti<'promise'>;
+  }
+
+  protected exposeExtrinsics (_extrinsics: SubmittableExtrinsics<'rxjs'>): SubmittableExtrinsics<'promise'> {
+    const creator = this.exposeMethod(_extrinsics);
+
+    return Object.keys(_extrinsics).reduce((result, sectionName) => {
+      const section = _extrinsics[sectionName];
+
+      result[sectionName] = Object.keys(section).reduce((result, methodName) => {
+        result[methodName] = this.exposeMethod(section[methodName]) as any;
```

Comment:
Not crazy at all about this list of stuff to do. This is where we are going to drop the ball. Same with storage, where it is even worse.